app/javascript/components/room_settings/RoomSettings.jsx

```diff
@@ -39,22 +37,53 @@ export default function RoomSettings() {
         <div className="mt-2">
           <Row>
             <Col className="border-end border-2">
-              <h6 className="text-primary">Room Naddme</h6>
-              <Button
-                variant="primary-light"
-                onClick={handleGenerateViewerAccessCode}
-              >
-                Generate
-              </Button>
-              {
-                accessCodes?.viewer_access_code
-              }
-              <Button
-                variant="danger"
-                onClick={handleDeleteViewerAccessCode}
-              >
-                Remove
-              </Button>
+              <Row>
+                <h6 className="text-primary">Room Name</h6>
+              </Row>
+              <Row>
+                <h6 className="text-primary">Generate access code for viewers</h6>
+                <div>
+                  <Button
+                    variant="primary-light"
+                    onClick={() => handleGenerateAccessCode({ role: 'Viewer' })}
```

Comment:
Why pass an object? 
`onClick={() => handleGenerateAccessCode('Viewer')}`

Reply:
I was asked to remove the arrow function last PR, I'm not sure I put it back in. Investigating this right now.